plugin/ale/sign.vim

```diff
@@ -40,6 +40,34 @@ execute 'sign define ALEWarningSign text=' . g:ale_sign_warning
 \   . ' texthl=ALEWarningSign'
 sign define ALEDummySign
 
+function! ale#sign#GetStatuslineStr()
+	" Returns a formatted string that can be integrated in the 
+	" statusline
+
+	redir => signs
+	silent exec 'sign place buffer=' . bufnr('%')
+	redir END
+
+	let [errors, warnings] = [0, 0]
+	for s in split(signs, "\n")
+		if s =~# 'ALEErrorSign$'
+			let errors += 1
+		elseif s =~# 'ALEWarningSign$'
+			let warnings += 1
+		endif
+	endfor
+
+	let errors = errors ?
+				\	printf(g:ale_statusline_format[0], errors) : ''
+	let warnings = warnings ?
+				\	printf(g:ale_statusline_format[1], warnings) : ''
```

Comment:
If these expressions are going off to the right margin anyway, you might as write only one line each here.